src/main/java/org/opentripplanner/netex/mapping/DatedServiceJourneyMapper.java

```diff
@@ -0,0 +1,125 @@
+package org.opentripplanner.netex.mapping;
+
+import org.opentripplanner.model.TripAlteration;
+import org.opentripplanner.model.TripAlterationOnDate;
+import org.opentripplanner.model.calendar.ServiceDate;
+import org.opentripplanner.netex.loader.support.HierarchicalMapById;
+import org.rutebanken.netex.model.DatedServiceJourney;
+import org.rutebanken.netex.model.DatedServiceJourneyRefStructure;
+import org.rutebanken.netex.model.EntityStructure;
+import org.rutebanken.netex.model.JourneyRefStructure;
+import org.rutebanken.netex.model.OperatingDay;
+import org.rutebanken.netex.model.ServiceAlterationEnumeration;
+import org.rutebanken.netex.model.ServiceJourneyRefStructure;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nullable;
+import javax.validation.constraints.NotNull;
+import javax.xml.bind.JAXBElement;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static org.rutebanken.netex.model.ServiceAlterationEnumeration.PLANNED;
+
+public class DatedServiceJourneyMapper {
+
+  private static final Logger LOG = LoggerFactory.getLogger(DatedServiceJourneyMapper.class);
+
+
+  /** Map of {@link TripAlteration} by service date, by trip id.  */
+  private final Map<String, SjAlt> resultById = new HashMap<>();
+  private final Map<String, DatedServiceJourney> dsjs;
+  private final HierarchicalMapById<OperatingDay> operatingDaysById;
+
+  public DatedServiceJourneyMapper(
+      HierarchicalMapById<DatedServiceJourney> dsjs,
+      HierarchicalMapById<OperatingDay> operatingDaysById
+  ) {
+    this.dsjs = dsjs.values().stream().collect(Collectors.toMap(EntityStructure::getId, Function.identity()));
+    this.operatingDaysById = operatingDaysById;
+  }
+
+  static Map<String, Map<ServiceDate, TripAlterationOnDate>> map(
+      HierarchicalMapById<DatedServiceJourney> dsjs,
+      HierarchicalMapById<OperatingDay> operatingDaysById
+  ) {
+    return new DatedServiceJourneyMapper(dsjs, operatingDaysById).map();
+  }
+
+  /**
+   * Index of alteration service-journey id and by service date.
+   */
+  private Map<String, Map<ServiceDate,TripAlterationOnDate>> map() {
+    for (String id : dsjs.keySet()) {
+      map(id);
+    }
+    return mapInternalResult();
+  }
+
+  private Map<String, Map<ServiceDate, TripAlterationOnDate>> mapInternalResult() {
```

Comment:
I see that you end up with TripAlterations by SJ id and ServiceDate, but how is this handled if there are duplicates? It looks like TripAlterations can replace other TripAlterations.